### PR TITLE
[GS-710] Use the Help Center API for multibrand article search

### PR DESCRIPTION
**Pull request by ellimist**
Stop querying the Support Search API and query the Help Center API instead.

## CCs
@zendesk/vegemite 
@zendesk/pegasus

### CODE REVIEW

**Comment by ellimist** [comment]
Pushed another change, to filter down only to the enabled help centers, since only those get suggested in the dropdown.

**Comment by ellimist** [comment]
Mandatory Screenshots:

Screenshot 2019-08-27 at 11 31 29
Screenshot 2019-08-27 at 11 31 15
Screenshot 2019-08-27 at 11 31 05
Screenshot 2019-08-27 at 11 30 48

**Comment by ellimist** [comment]
CC @zendesk/vegemite @zendesk/guide-search

File: app.js [hunk 1]
```diff
@@ -70,12 +70,16 @@
         }
 
         if (this.isMultibrand) {
-          url = '/api/v2/search.json';
-          data.brand_id = this.$('.brand-filter').zdSelectMenu('value');
-          data.query = 'type:article ' + data.query;
-
-          if (data.brand_id !== 'any') {
-            data.query = 'brand:' + data.brand_id + ' ' + data.query;
+          data.multibrand = true
```
Comment (Jiyu-Zhang-Zendesk) [inline app.js]: better to have semi colon here

File: app.js [hunk 2]
```diff
@@ -70,12 +70,16 @@
...
+          var selectedBrand = this.$('.brand-filter').zdSelectMenu('value');
+
+          if (selectedBrand === 'any') {
+            // Build a comma separated string of brand ids to filter down to
+            // otherwise HC will search in the restricted help centers too
+            // if the current user has permission to access them
+            data.brand_id = this.getPublicBrandIDs().join(",");
```
Comment (Jiyu-Zhang-Zendesk) [inline app.js]: we can use single quotes here

Comment (Jiyu-Zhang-Zendesk) [reply app.js]: do we want to handle that `getPublicBrandIDs()` is empty?

File: app.js [hunk 3]
```diff
@@ -70,12 +70,16 @@
...
+          } else {
+            data.brand_id = selectedBrand
```
Comment (Jiyu-Zhang-Zendesk) [inline app.js]: better to have semi colon here

File: app.js [hunk 4]
```diff
@@ -104,8 +108,8 @@
         var brandsData = brandsResponse[0],
             localeData = localeResponse[0];
 
-        var brands = this.filterBrands(brandsData.brands);
-        this.isMultibrand = brands.length > 1;
+        this.publicBrands = this.filterBrands(brandsData.brands);
```
Comment (Jiyu-Zhang-Zendesk) [inline app.js]: do you want to set default value for `this.publicBrands` before ajax (like [])?

File: app.js [hunk 5]
```diff
@@ -379,5 +383,11 @@
         return element.active && element.help_center_state === "enabled";
       });
     },
+
+    getPublicBrandIDs: function(){
+      return _.map(this.publicBrands, function(brand) {
+        return brand.id
```
Comment (Jiyu-Zhang-Zendesk) [inline app.js]: better to have semi colon here

**Comment by ellimist** [comment]
@zendesk/pegasus

**Pull request merged**